tests/test_models.py

```diff
@@ -224,23 +240,41 @@ def test_billing_agent_save(self):
         self.assertFalse(other_has_country)
         self.assertFalse(default_still_default)
 
-    def test_currency_exchange_rate(self):
-        with patch(
-            'plugins.consortial_billing.logic.latest_multiplier_for_indicator'
-        ) as latest_multiplier:
-            self.currency_other.exchange_rate()
-            self.assertIn(
-                plugin_settings.RATE_INDICATOR,
-                latest_multiplier.call_args.args,
-            )
-            self.assertIn(
-                self.currency_other.region,
-                latest_multiplier.call_args.args,
-            )
-            self.assertIn(
-                self.currency_other.region,
-                latest_multiplier.call_args.args,
-            )
+    @patch('plugins.consortial_billing.logic.latest_multiplier_for_indicator')
+    def test_currency_exchange_rate_with_typical_args(self, latest_multiplier):
+        self.currency_base.exchange_rate(base_band=self.band_base_country_other)
+        self.assertIn(
+            plugin_settings.RATE_INDICATOR,
+            latest_multiplier.call_args.args,
+        )
+        # Target currency
+        self.assertIn(
+            self.currency_base.region,
+            latest_multiplier.call_args.args,
+        )
+        # Base currency generated with specified base band
+        self.assertIn(
+            self.currency_other.region,
+            latest_multiplier.call_args.args,
+        )
+
+    @patch('plugins.consortial_billing.logic.latest_multiplier_for_indicator')
+    def test_currency_exchange_rate_with_no_args(self, latest_multiplier):
+        self.currency_other.exchange_rate()
+        self.assertIn(
+            plugin_settings.RATE_INDICATOR,
+            latest_multiplier.call_args.args,
+        )
+        # Target currency
+        self.assertIn(
+            self.currency_other.region,
+            latest_multiplier.call_args.args,
+        )
+        # Base currency generated with default base band
+        self.assertIn(
+            self.currency_other.region,
+            latest_multiplier.call_args.args,
+        )
```

Comment:
Ideally a unit test like this one should have only one assertion. In this case they could be grouped into one membership check of your arguments into `latest_multiplier.call_args.args` for each of the two tests

Reply:
Thanks @mauromsl. There's a limit to the one-assertion-per-test premise but it does help in this case. I've condensed these two tests.